src/Hosting/Hosting/test/HostingApplicationTests.cs

```diff
@@ -87,7 +88,81 @@ public void CreateContextCreatesNewContextIfNotUsingDefaultHttpContextFactory()
             hostingApplication.DisposeContext(context, null);
         }
 
-        private static HostingApplication CreateApplication(IHttpContextFactory httpContextFactory = null, bool useHttpContextAccessor = false)
+        [Fact]
+        public void IHttpActivityFeatureIsCreated()
```

Comment:
Add a test without an activity source and validate that `IHttpActivityFeature.Activity` is null.